File: skill-chain-backend/src/main/java/com/skillchain/entity/Schedule.java
package com.skillchain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("schedule")
public class Schedule {
    @TableId(type = IdType.AUTO)
    private Long scheduleId;

    private Long providerId;

    private Long skillId;

    private LocalDate date;

    private String timeSlot;

    private Integer status;

    private String location;

    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: skill-chain-frontend/src/api/schedule.ts
import request from '@/utils/request'

export interface ScheduleItem {
  scheduleId?: number
  providerId?: number
  skillId?: number | null
  date?: string
  timeSlot?: string
  status?: number   // 0=空闲 1=已预约 2=锁定
  location?: string
}

export const getMySchedules = (date?: string) =>
  request.get<ScheduleItem[]>('/schedule/my', { params: date ? { date } : {} })

export const addSchedule = (data: { date: string; timeSlot: string }) =>
  request.post('/schedule/add', data)

export const deleteSchedule = (id: number) =>
  request.delete(`/schedule/${id}`)
